pandora-common/src/utility.ts

```diff
@@ -200,6 +200,36 @@ export function FormatTimeInterval(time: number, mode: 'full' | 'short' = 'full'
 	return res;
 }
 
+export type TimeUnit = 'milliseconds' | 'seconds' | 'minutes' | 'hours' | 'days' | 'weeks' | 'months' | 'years';
+
+/**
+ * Converts a time value in specified unit to milliseconds
+ * @param time The time value
+ * @param unit The unit of the time value
+ * @returns The time value in milliseconds
+ * @note This function is not precise for months and years as they have varying length
+ */
+export function TimeSpanMs(time: number, unit: TimeUnit): number {
+	switch (unit) {
+		case 'milliseconds':
+			return time;
+		case 'seconds':
+			return time * 1000;
+		case 'minutes':
+			return time * 1000 * 60;
+		case 'hours':
+			return time * 1000 * 60 * 60;
+		case 'days':
+			return time * 1000 * 60 * 60 * 24;
+		case 'weeks':
+			return time * 1000 * 60 * 60 * 24 * 7;
+		case 'months':
+			return time * 1000 * 60 * 60 * 24 * 30;
+		case 'years':
+			return time * 1000 * 60 * 60 * 24 * 365;
```

Comment:
I don't think we want to allow `months`/`years` - they are both imprecise - it would be better to define the values in code in days if such high values are wanted (same as we have reverse support for in `FormatTimeInterval`)